core/chains/evm/log/broadcaster.go

```diff
@@ -404,12 +404,13 @@ func (b *broadcaster) onNewLog(log types.Log) {
 	b.maybeWarnOnLargeBlockNumberDifference(int64(log.BlockNumber))
 
 	if log.Removed {
-		b.logPool.removeLog(log)
+		// Remove the whole block that contained this log.
+		b.logPool.RemoveBlock(log.BlockHash, log.BlockNumber)
 		return
 	} else if !b.registrations.isAddressRegistered(log.Address) {
 		return
 	}
-	if b.logPool.addLog(log) {
+	if b.logPool.AddLog(log) {
```

Comment:
Why has this been made public?

Reply:
I made the interface name private.
The internal methods, unless made public, can't be accessed from unit-tests which are in _test package.

Do you know any other way to export all private methods just for testing?

Reply:
Yup, put them in `exports_test.go` or `helpers_test.go`